misc/snapcraft_releases.py

```diff
@@ -0,0 +1,109 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+import logging
+from argparse import ArgumentParser
+from enum import StrEnum
+
+from releaser import (  # type: ignore (pyright struggles with this when run from server folder)
+    Version,
+)
+
+log = logging.getLogger(__name__)
+
+
+class Track(StrEnum):
+    Latest = "latest"
+    V2 = "v2"
+    V3 = "v3"
+    Nightly = "nightly"
+
+
+class Channel(StrEnum):
+    Stable = "stable"
+    Candidate = "candidate"
+    Beta = "beta"
+    Edge = "edge"
+
+
+CHANNELS = (Channel.Stable, Channel.Candidate, Channel.Beta, Channel.Edge)
+
+
+def get_tracks_for_version(version: Version) -> list[Track]:
+    TRACKS_FOR_V2 = (Track.V2,)
+    # TODO: Add Latest track once we publicly release v3
+    TRACKS_FOR_V3 = (Track.V3,)
+
+    tracks: list[Track] = []
+    match version.major:
+        case 2:
+            tracks.extend(TRACKS_FOR_V2)
+        case 3:
+            tracks.extend(TRACKS_FOR_V3)
+        case _:
+            raise ValueError(f"Unsupported major version: {version.major}")
+    return tracks
+
+
+assert get_tracks_for_version(Version.parse("2.0.0")) == [Track.V2]
+assert get_tracks_for_version(Version.parse("3.0.0")) == [Track.V3]
+
+
+def get_channel_for_version(version: Version) -> Channel:
+    PRERELEASE_TO_CHANNEL = {
+        "a": Channel.Edge,
+        "b": Channel.Beta,
+        "rc": Channel.Candidate,
+    }
+
+    if version.prerelease:
+        for pre, chan in PRERELEASE_TO_CHANNEL.items():
+            if version.prerelease.startswith(pre):
+                return chan
+        raise ValueError(f"Unsupported prerelease: {version.prerelease}")
+    elif version.dev is not None:
+        return Channel.Edge
+    else:
+        return Channel.Stable
+
+
+assert get_channel_for_version(Version.parse("3.0.0")) == Channel.Stable
+assert get_channel_for_version(Version.parse("3.0.0-a.0")) == Channel.Edge
+assert get_channel_for_version(Version.parse("3.0.0-b.0")) == Channel.Beta
+assert get_channel_for_version(Version.parse("3.0.0-rc.0")) == Channel.Candidate
+assert get_channel_for_version(Version.parse("3.0.0-dev.0")) == Channel.Edge
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        "snapcraft_channel",
+        description="""
+    Determine the snapcraft channel based on the provided options and version.
+    """,
+    )
```

Comment:
Script is named `snapcraft_releases`, I would change descriptions here:
```suggestion
    parser = ArgumentParser(
        "snapcraft_releases",
        description="""
    Output as many snapcraft's `--release=<channel>` options based on the given Parsec version.
    """,
        epilog="See https://snapcraft.io/docs/channels")
    )
```